src/goto-instrument/contracts/contracts.cpp

```diff
@@ -911,6 +911,9 @@ void code_contractst::instrument_call_statement(
   }
   else if(callee_name == "free")
   {
+    source_locationt &location_no_checks =
+      instruction_it->source_location_nonconst();
```

Comment:
Do you really want or need to take a reference here? The reference will ensure (but is that desired?) that checks are disabled on the function call. Otherwise, I would have expected that you'd be taking a copy here, at which point you want to use `source_location()` instead of `source_location_nonconst()`.

Reply:
fixed